app/routers/event.py

```diff
@@ -323,6 +332,63 @@ def add_new_event(values: dict, db: Session) -> Optional[Event]:
         return None
 
 
+def extract_shared_list_from_data(data: ImmutableMultiDict,
+                                  db: Session) -> Optional[SharedList]:
+    """Extract shared list items from POST data.
+    Return:
+        SharedList: SharedList object stored in the database.
+    """
+
+    Item = namedtuple("Item", ["name", "amount", "participant"])
+    shared_list_raw = {
+        'item_name': data.getlist('item_name'),
+        'item_amount': data.getlist('item_amount'),
+        'item_participant': data.getlist('item_participant')
+    }
+    shared_list = {"title": data.get("title"),
+                   "items": []}
+    for i in range(len(shared_list_raw.get('item_name'))):
```

Comment:
Please think of a better way to phrase this for loop (hint: what are you trying to loop over?)